src/snowflake/snowpark/dataframe_reader.py

```diff
@@ -1024,3 +1073,480 @@ def _read_semi_structured_file(self, path: str, format: str) -> DataFrame:
         df._reader = self
         set_api_call_source(df, f"DataFrameReader.{format.lower()}")
         return df
+
+    @private_preview(version="1.29.0")
+    def dbapi(
+        self,
+        create_connection: Callable[[], "Connection"],
+        table: str,
```

Comment:
Do we accept a sql query?

Reply:
not yet, we plan to do it in next phase

Reply:
query support is here: https://github.com/snowflakedb/snowpark-python/pull/3097